src/main/java/seedu/duke/reflection/ReflectionManager.java

```diff
@@ -0,0 +1,149 @@
+package seedu.duke.reflection;
+
+import seedu.duke.command.Command;
+import seedu.duke.exception.BadCommandException;
+import seedu.duke.command.CommandParser;
+import seedu.duke.exception.InvalidCommandException;
+import seedu.duke.manager.Manager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class ReflectionManager extends Manager {
+    private static final String FEATURE_NAME = "Self Reflection";
+    private static final String BRIEF_DESCRIPTION = "Users can get a random set of questions to reflect on.";
+    private static final String FULL_DESCRIPTION = "";
+    private static String commandType;
+    private static HashMap<String, String> argumentPayload;
+    private static final String GET_COMMAND = "get";
+    private static final String RETURN_MAIN = "return";
+    private static final String EXIT_COMMAND = "exit";
+    private static final CommandParser parser;
+    private ArrayList<Command> supportedCommands;
+    private ArrayList<Manager> supportedManagers;
+
+    public ReflectionManager() {
+        this.parser = getCommandParser();
+        this.supportedCommands = getSupportedCommands();
+        this.supportedManagers = getSupportedFeatureManagers();
+    }
+
+    /**
+     * Get Self Reflection feature name.
+     *
+     * @return Self Reflection feature name
+     */
+    @Override
+    public String getFeatureName() {
+        return FEATURE_NAME;
+    }
+
+    /**
+     * Get a summary of description of self reflection feature.
+     *
+     * @return Brief description of self reflection
+     */
+    @Override
+    public String getBriefDescription() {
+        return BRIEF_DESCRIPTION;
+    }
+
+    /**
+     * Get a full description of self reflection feature.
+     * <br/>
+     * FULL_DESCRIPTION is not completed yet!
+     *
+     * @return Full description of self reflection
+     */
+    @Override
+    public String getFullDescription() {
+        return FULL_DESCRIPTION;
+    }
+
+    @Override
+    protected void setSupportedCommands() {
+        supportedCommands.add();
+    }
+
+    @Override
+    protected void setSupportedFeatureManagers() {
+
+    }
+
+    @Override
+    public void runEventDriver() {
+        SelfReflection selfReflection = new SelfReflection();
+        selfReflection.run();
+    }
```

Comment:
Does this compile? There is no constructor for SelfReflection.

Reply:
I think it will just auto generate an empty constructor? I'm not too sure but I shall add constructor